src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -52,6 +52,7 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 {
                     // Delay register the url so users don't accidently overwrite it.
                     hostBuilder.UseSetting(WebHostDefaults.ServerUrlsKey, address);
+                    hostBuilder.PreferHostingUrls(true);
```

Comment:
This is only for when you run behind ANCM and starting the app on localhost using the port specified `ASPNETCORE_PORT`. We don't want the default template to PreferHostingUrls. We need to use this default since UseIISIntegration is expected to be injected via the new hosting assembly startup feature.